src/components/rpc_base/include/rpc_base/rpc_base_json_inl.h

```diff
@@ -287,6 +287,61 @@ Json::Value Array<T, minsize, maxsize>::ToJsonValue() const {
   return array;
 }
 
+////////// Array to be instantiated with Enum<T>
+
+// Non-const version
+template <typename E, size_t minsize, size_t maxsize>
+Array<Enum<E>, minsize, maxsize>::Array(Json::Value* value)
+    : CompositeType(InitHelper(value, &Json::Value::isArray)) {
+  if (value) {
+    if (value->isArray()) {
+      this->reserve(value->size());
+      for (Json::Value::iterator i = value->begin(); i != value->end(); ++i) {
+        Enum<E> item(&*i);
+        if (!item.is_valid()) {
+          continue;
+        }
+        push_back(item);
+      }
+    } else {
+      // Array is empty, empty initialized or uninitialized
+      // depending on InitHelper result
+    }
+  }
+}
+
+// Const version, must be identical to the non-const version
+template <typename E, size_t minsize, size_t maxsize>
+Array<Enum<E>, minsize, maxsize>::Array(const Json::Value* value)
+    : CompositeType(InitHelper(value, &Json::Value::isArray)) {
+  if (value) {
```

Comment:
@LuxoftAKutsan is it possible to use const version of ctor in non-const ctor instead of code duplication? Especially if they must be identical

Reply:
@AKalinich-Luxoft Fixed in d1308970ca2afad7de251f6ef744d94c4422cb42